firmware/shared/src/io/io_loadswitch.c

```
@@ -0,0 +1,44 @@
+#include "io_loadswitch.h"
+#include "hw_gpio.h"
+
+#define ADC_VOLTAGE_TO_CURRENT_A 1.720f
+
+void io_loadswitch_setChannel(const Efuse *channel, const bool enabled)
+{
+    if (channel->enable_gpio != NULL)
+    {
+        hw_gpio_writePin(channel->enable_gpio, enabled);
+    }
+}
+
+bool io_loadswitch_isChannelEnabled(const Efuse *channel)
+{
+    return hw_gpio_readPin(channel->enable_gpio);
+}
+
+float io_loadswitch_getChannelCurrent(const Efuse *channel)
+{
+    const AdcChannel *current_sense = channel->sns_adc_channel;
+    if (current_sense == NULL)
```

surely you assert that it exists? otherwise silent fail

---

Yeah this should be a asset coz if current_sense =  null thats kind of a skill issue